File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/2095-minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        Stack<Character> st = new Stack();
        int cnt = 0;
        for(Character c : s.toCharArray()){
            if(c == '[') st.push(c);
            else{
                if(st.isEmpty()) cnt++;
                else st.pop();
            }
        }
        return (cnt + 1)/2;
    }
}

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/2095-minimum-number-of-swaps-to-make-the-string-balanced.cpp
class Solution {
public:
    int minSwaps(string s) {
        stack<char> st;
        int unbalanced = 0;
        for(char c : s){
            if(c == '[') st.push(c);
            else{
                if(!st.empty()) st.pop();
                else unbalanced++;
            }
        }
        return (unbalanced+1) / 2;
    }
};